build.sh

```diff
@@ -71,6 +78,19 @@ EOF
     mkdir -p "$SCRIPTDIR/blake2bf/build/${OSARCH}/lib"
     mv libblake2bf.so "$SCRIPTDIR/blake2bf/build/${OSARCH}/lib"
   fi
+  if [[ $(uname -m) == 'arm64' ]]; then
```

Comment:
my understanding is this adds support for building on ARM based MAC, right? 
it seems that this part could be easily merged with the linux part, extending the if to include `"$OSTYPE" == "darwin"*` 

Moreover if it possible to swap `arm64` dir with `aarch64` dir, there should be no need to check the architecture

Reply:
It doesn't create a dylib, so it's not a complete mac build.

Reply:
This adds support for arm build on mac m1, which is the besu runner. I wasn't sure linux arm64 was compatible with apple aarch64.

Reply:
I do not have access to a Mac M1 to verify if the script could be generalized in the way I proposed in this PR https://github.com/atoulme/besu-native/pull/3, it will be great if you can try it